### PR TITLE
[WIP]Enable Citadel Agent run as nonroot

### PR DESCRIPTION
**Pull request by clyang82**
Signed-off-by: Chun Lin Yang <clyang@cn.ibm.com>

This PR is for non-root user support for citadel. refer to https://github.com/istio/istio/issues/10768

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[x] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by clyang82** [comment]
nodeagent seems cannot run as nonroot, because nodeagent listen the unix domain socket
https://github.com/istio/istio/blob/b373417168adfa83e6b84a141e7ff7c13b96c746/security/pkg/nodeagent/sds/server.go#L278:289

**Comment by myidpt** [comment]
Thanks, @clyang82 . Please check whether the non-root process can work with UDS.

File: security/pkg/nodeagent/sds/server.go [hunk 1]
```diff
@@ -294,14 +295,23 @@ func (s *Server) initGatewaySdsService(options *Options) error {
...
 	// Remove unix socket before use.
-	if err := os.Remove(udsPath); err != nil && !os.IsNotExist(err) {
+	if err = os.Remove(udsPath); err != nil && !os.IsPermission(err) && !os.IsNotExist(err) {
 		// Anything other than "file not found" is an error.
 		sdsServiceLog.Errorf("Failed to remove unix://%s: %v", udsPath, err)
 		return nil, fmt.Errorf("failed to remove unix://%s", udsPath)
 	}
+	// if no permission, try to run command with sudo
+	if os.IsPermission(err) {
+		cmd := exec.Command("/bin/sh", "-c", "sudo rm -r "+udsPath)
```
[the remark this answers is not in the document]
Comment (clyang82) [reply security/pkg/nodeagent/sds/server.go]: @myidpt Ideally, the image should allow running the container with any uid. there has some security concerns about running as `root`.

File: security/docker/Dockerfile.node-agent [hunk 2]
```diff
@@ -7,9 +7,14 @@ ARG BASE_VERSION=latest
 # The following section is used as base image if BASE_DISTRIBUTION=default
 FROM docker.io/istio/base:${BASE_VERSION} as default
 
+# Sudoers used to allow debug utilities.
```
[the remark this answers is not in the document]
Comment (clyang82) [reply security/docker/Dockerfile.node-agent]: @howardjohn could you point me how to change it in base image? As I mentioned in another PR, if I change the base image and then make the changes accordingly in a PR. in the testing, it is still using the old base image. Thanks.

[the remark this answers is not in the document]
Comment (clyang82) [reply security/docker/Dockerfile.node-agent]: base image was updated. #17255

File: security/pkg/nodeagent/sds/server.go [hunk 3]
```diff
@@ -313,10 +323,19 @@ func setUpUds(udsPath string) (net.Listener, error) {
 		sdsServiceLog.Errorf("SDS uds file %q doesn't exist", udsPath)
 		return nil, fmt.Errorf("sds uds file %q doesn't exist", udsPath)
 	}
-	if err := os.Chmod(udsPath, 0666); err != nil {
+	if err = os.Chmod(udsPath, 0666); err != nil && !os.IsPermission(err) {
```
[the remark this answers is not in the document]
Comment (clyang82) [reply security/pkg/nodeagent/sds/server.go]: I think we still need to have read/write permission. any comments?

File: security/pkg/nodeagent/sds/server.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (howardjohn) [reply security/pkg/nodeagent/sds/server.go]: But if anyone can just run `sudo` what is the difference between root and not root?

File: security/pkg/nodeagent/sds/server.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (lei-tang) [reply security/pkg/nodeagent/sds/server.go]: E.g., will 0660 or 0600 also work? Just try to be as restrictive as possible.

**Pull request closed without merging**